app/controllers/downloads_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# DownloadsController: for handling user downloads
+class DownloadsController < ApplicationController
+  before_action :authenticate_user!
+
+  def download
+    redirect_to root_url unless current_user.can_export?
+    lookup = params[:lookup]
+    if lookup.blank?
+      respond_to do |format|
+        format.any { head :bad_request }
+      end
+      return
+    end
+
+    download = current_user.downloads.find_by(lookup: lookup)
+    if download.nil?
+      respond_to do |format|
+        format.any { head :not_found }
+      end
+      return
+    end
+
+    send_data(Base64.decode64(download.contents), filename: download.filename) && download.destroy
```

Comment:
instead of destroying immediately, thoughts on destroying after maybe 24 hours of the file being generated? in case the user has trouble downloading the first time, maybe wants to download on multiple devices (not sure if users would do this), or send the link to someone else to download as well?

Reply:
I can also imagine that sometimes users might not end up downloading the file for a while (maybe because they are just playing with the export feature or something), which will lead to this file data sitting in the db indefinitely

Reply:
Sitting indefinitely is fine, these files have size limits, and only one per type per user account is allowed. If we do eventually see this an an error, we can add an additional job that goes through periodically and deletes them. I do want to be strict about immediate deletion, and we definitely don't want people to be able to share links. It's a per user account download.